Source/FlowEditor/Private/DetailCustomizations/FlowNode_CustomNodeBaseDetails.cpp

```diff
@@ -0,0 +1,152 @@
+		.NameContent()
+			[
+				SNew(STextBlock)
+					.Text(EventNameText)
+			]
+		.ValueContent()
+			.HAlign(HAlign_Fill)
+			[
+				SAssignNew(EventTextListWidget, SComboBox<TSharedPtr<FName>>)
+					.OptionsSource(&EventNames)
+					.OnGenerateWidget(this, &FFlowNode_CustomNodeBaseDetails::GenerateEventWidget)
+					.OnComboBoxOpening(this, &FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening)
+					.OnSelectionChanged(this, &FFlowNode_CustomNodeBaseDetails::PinSelectionChanged)
+					[
+						SNew(STextBlock)
+							.Text(this, &FFlowNode_CustomNodeBaseDetails::GetSelectedEventText)
+					]
+			];
+}
+
+void FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening()
+{
+	RebuildEventNames();
+}
+
+void FFlowNode_CustomNodeBaseDetails::RebuildEventNames()
+{
+	EventNames.Empty();
+
+	check(CachedEventNameSelected.IsValid());
+	EventNames.Add(CachedEventNameSelected);
+
+	if (ObjectsBeingEdited[0].IsValid() && ObjectsBeingEdited[0].Get()->GetOuter())
+	{
+		const UFlowAsset* FlowAsset = CastChecked<UFlowAsset>(ObjectsBeingEdited[0].Get()->GetOuter());
+		TArray<FName> SortedNames = BuildEventNames(*FlowAsset);
+
+		if (bExcludeReferencedEvents)
+		{
+			for (const TPair<FGuid, UFlowNode*>& Node : FlowAsset->GetNodes())
+			{
+				if (Node.Value->GetClass()->IsChildOf(UFlowNode_CustomNodeBase::StaticClass()))
+				{
+					SortedNames.Remove(Cast<UFlowNode_CustomNodeBase>(Node.Value)->GetEventName());
+				}
+			}
+		}
+
+		SortedNames.Sort([](const FName& A, const FName& B)
+		{
+			return A.LexicalLess(B);
+		});
+
+		for (const FName& EventName : SortedNames)
+		{
+			const bool bIsCurrentSelection = (EventName == *CachedEventNameSelected);
+			if (!EventName.IsNone() && !bIsCurrentSelection)
+			{
+				EventNames.Add(MakeShared<FName>(EventName));
+			}
+		}
+	}
+
+	if (!IsInEventNames(NAME_None))
+	{
+		EventNames.Add(MakeShared<FName>(NAME_None));
+	}
+}
+
+bool FFlowNode_CustomNodeBaseDetails::IsInEventNames(const FName& EventName) const
+{
+	const bool bIsInEventNames =
+		EventNames.ContainsByPredicate([&EventName](const TSharedPtr<FName>& ExistingName)
+			{
+				return *ExistingName == EventName;
+			});
+
+	return bIsInEventNames;
+}
+
+TSharedRef<SWidget> FFlowNode_CustomNodeBaseDetails::GenerateEventWidget(const TSharedPtr<FName> Item) const
+{
+	return
+		SNew(STextBlock)
+		.Text(FText::FromName(*Item.Get()));
+}
+
+FText FFlowNode_CustomNodeBaseDetails::GetSelectedEventText() const
+{
+	check(CachedEventNameSelected.IsValid());
+
+	return FText::FromName(*CachedEventNameSelected.Get());
+}
+
+void FFlowNode_CustomNodeBaseDetails::PinSelectionChanged(const TSharedPtr<FName> Item, ESelectInfo::Type SelectInfo)
+{
+	ensure(ObjectsBeingEdited[0].IsValid());
+
+	UFlowNode_CustomNodeBase* Node = Cast<UFlowNode_CustomNodeBase>(ObjectsBeingEdited[0].Get());
+	if (IsValid(Node) && Item)
+	{
+		const bool bIsChanged = (*CachedEventNameSelected != *Item);
+
+		if (bIsChanged)
+		{
+			CachedEventNameSelected = Item;
```

Comment:
_We need to cache the selected item, and it also needs to be a member of the EventNames list (SComboBox code assumes that it is, and specifically the same TSharedPtr object, not just the value being the same)._